manifests/010-prometheus-rules.yaml

```diff
@@ -9,14 +9,30 @@ spec:
   groups:
   - name: SamplesOperator
     rules:
+    - alert: SamplesRetriesMissingOnImagestreamImportFailing
+      expr: sum(openshift_samples_failed_imagestream_import_info) > 0 && openshift_samples_retry_imagestream_import_total <= openshift_samples_retry_imagestream_import_total offset 30m && sum(openshift_samples_invalidsecret_info) == 0
+      for: 2h
+      labels:
+        severity: warning
+      annotations:
+        message: |
+          Samples operator is detecting problems with imagestream image imports, and the periodic retries of those imports are not occurring.  Contact support.  You can look at the "openshift-samples" ClusterOperator object for details. You can also query 'openshift_samples_failed_imagestream_import_info' to see which ImageStreams do not import. Most likely there are issues with the external image registry hosting the images that need to be investigated.  If registry.redhat.io is used, then samples operator has at least determined that pull credentials exist, and that network connections to registry.redhat.io were previously successful.
```

Comment:
Could we use alerting rule templates here [1]? Could we surface the individual ImageStreams that are failing to import?

I am a bit loathe to instruct admins to query a metric when the alerting rules have access to the same info.

[1] https://prometheus.io/docs/prometheus/latest/configuration/template_examples/

Reply:
The surface individual imagestreams with the underlying metric and hence corresponding alert was an earlier discussion point of ours as I'm sure you remember, where we went with coarse grained / do not include specific imagestream(s).

Not sure I'm inclined to change course on that yet.  Among other things, I'm not sure exactly how to do it yet.

I'll read up on templates in the interim.  Feel free to elaborate on the "how" while I do that.

Reply:
So I spent a day trying to produce an alert @adambkaplan with the current form of the metric / query but with no luck

I think I have a new query which should work.  Will try that out tomorrow.  Assuming I have luck, I'll then see about changing the new retry metric to a CounterVec to see about adding labels, etc. for use in templates.